libs/vm-modules/tests/unit/ml/ml_model_tests.cpp

```diff
@@ -1137,6 +1165,254 @@ TEST_F(VMModelTests, model_sequential_flatten_2d_in_2d_out)
   EXPECT_TRUE(constructed_shape == expected.shape());
 }
 
+TEST_F(VMModelTests, model_sequential_reshape_2d_in_2d_out)
+{
+  static char const *SRC_METRIC = R"(
+          function main() : Tensor
+                          var shape = Array<UInt64>(1);
+                          shape[0] = 1u64;
+                          var x = Tensor(shape);
+                          var str_vals = "0.5; 7.1; 9.1; 6.2;";
+                          x.fromString(str_vals);
+
+                          var to_shape = Array<UInt64>(2);
+                          to_shape[0] = 4u64;
+                          to_shape[1] = 1u64;
+
+                          var model = Model("sequential");
+                          model.add("reshape", to_shape);
+                          model.compile("scel", "adam");
+                          var prediction = model.predict(x);
+
+                          return prediction;
+          endfunction
+      )";
+
+  Variant res;
+  ASSERT_TRUE(toolkit.Compile(SRC_METRIC));
+  ASSERT_TRUE(toolkit.Run(&res, ChargeAmount{0}));
+  auto const prediction        = res.Get<Ptr<fetch::vm_modules::math::VMTensor>>();
+  auto const constructed_shape = prediction->shape();
+
+  fetch::math::Tensor<fetch::vm_modules::math::DataType> gt({4, 1});
+  gt(0, 0) = fetch::math::Type<DataType>("+0.5");
+  gt(1, 0) = fetch::math::Type<DataType>("+7.1");
+  gt(2, 0) = fetch::math::Type<DataType>("+9.1");
+  gt(3, 0) = fetch::math::Type<DataType>("+6.2");
+  ASSERT_TRUE((prediction->GetTensor())
+                  .AllClose(gt, fetch::math::function_tolerance<DataType>(),
+                            fetch::math::function_tolerance<DataType>()));
+  EXPECT_TRUE(constructed_shape == gt.shape());
+}
+
+TEST_F(VMModelTests, model_sequential_reshape_2d_in_2d_out_wrong_shape)
+{
+  static char const *SRC_METRIC = R"(
+          function main() : Tensor
+                          var shape = Array<UInt64>(1);
+                          shape[0] = 1u64;
+                          var x = Tensor(shape);
+                          var str_vals = "0.5; 7.1; 9.1; 6.2;";
+                          x.fromString(str_vals);
+
+                          var to_shape = Array<UInt64>(2);
+                          to_shape[0] = 1u64;
+                          to_shape[1] = 4u64;
+
+                          var model = Model("sequential");
+                          model.add("reshape", to_shape);
+                          model.compile("scel", "adam");
+                          var prediction = model.predict(x);
+
+                          return prediction;
+          endfunction
+      )";
+
+  ASSERT_TRUE(toolkit.Compile(SRC_METRIC));
+  ASSERT_FALSE(toolkit.Run(nullptr, ChargeAmount{0}));
```

Comment:
why does this fail? 1, 4 has the same number of elements as the input string right?

Reply:
{4,1} != {1,4}

Reply:
so why doesn't it just reshape your 4,1 tensor into a 1,4 tensor? I don't see why it fails to run?

Reply:
it is an attempt to reshape a tensor of shape 1 by batch_dim into a 4 by batch_dim and batch dim should'n be changed